src/compiler/binder.rs

```diff
@@ -115,15 +115,26 @@ impl BinderType {
         self.Symbol()(flags, name)
     }
 
-    fn add_declaration_to_symbol(&self, symbol: Rc<Symbol>, node: Rc<Node /*Declaration*/>) {
+    fn add_declaration_to_symbol(
+        &self,
+        symbol: Rc<Symbol>,
+        node: Rc<Node /*Declaration*/>,
+        symbol_flags: SymbolFlags,
+    ) {
+        symbol.set_flags(symbol.flags() | symbol_flags);
+
         node.set_symbol(symbol.clone());
         let declarations = {
             append_if_unique(
                 symbol.maybe_declarations().as_ref().map(|vec| vec.clone()),
-                node,
+                node.clone(),
             )
         };
         symbol.set_declarations(declarations);
+
+        if symbol_flags.intersects(SymbolFlags::Value) {
+            set_value_declaration(&*symbol, node);
```

Comment:
`symbol.value_declaration` is going to be used by the type-checking so wiring that up here in the binder